test/it/parser/src/main/resources/case/rdl/create.xml

```diff
@@ -615,6 +647,34 @@
             <key-generate-strategy algorithm-name="snowflake" />
         </rule>
     </create-sharding-table-rule>
+    
+    <create-sharding-table-rule sql-case-id="create-sharding-table-rule-with-complex-table-strategy-and-standard-database-strategy">
+        <rule name="t_order_item">
+            <data-node>ds_${0..1}.t_order_item_${0..1}</data-node>
+            <table-strategy type="complex" sharding-column="order_id">
+                <algorithm-segment algorithm-name="complex_inline">
+                    <property key="algorithm-expression" value="t_order_${user_id % 2}" />
+                </algorithm-segment>
+            </table-strategy>
+            <database-strategy type="standard" sharding-column="user_id">
+                <algorithm-segment algorithm-name="inline">
+                    <property key="algorithm-expression" value="t_order_${user_id % 2}" />
```

Comment:
The algorithm-expression does not reflect the characteristics of complex

Reply:
@Qianyi951015 FYI:
```sql
CREATE SHARDING TABLE RULE t_order_item (
DATANODES("ds.t_order_${0..1}_${0..1}"),
TABLE_STRATEGY(TYPE=COMPLEX,SHARDING_COLUMNS=user_id,order_id,SHARDING_ALGORITHM(TYPE(NAME=COMPLEX_INLINE,PROPERTIES("algorithm-expression"="t_order_${user_id % 2}_${order_id % 2}"))))
);
```

Reply:
fixed